packages/emotion/src/InstUISettingsProvider/index.tsx

```diff
@@ -0,0 +1,104 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import React, { useContext } from 'react'
+import { TextDirectionContext } from '@instructure/ui-i18n'
+import { ThemeProvider } from '@emotion/react'
+import { ThemeOrOverride } from '../EmotionTypes'
+import { getTheme } from '../EmotionThemeProvider'
+
+type InstUIProviderProps = {
+  theme?: ThemeOrOverride
+  dir?: 'ltr' | 'rtl' // TODO allow "auto" too
+}
+/**
+ * ---
+ * category: components/utilities
+ * ---
+ * Wrapper for emotion js's [ThemeProvider](https://emotion.sh/docs/theming#themeprovider-reactcomponenttype).
+ *
+ * Applies the given theme. It handles either a full theme, or an overrides object.
+ * You can also specify the requested text direction for its descendants
+ *
+ * ```js
+ * import { canvas, instructure } from '@instructure/ui-themes'
+ * import { InstUISettingsProvider } from '@instructure/emotion'
+ *
+ * @example
+ * <InstUISettingsProvider theme={canvas}>
+ *   <div>Canvas themed part</div>
+ *
+ *   <InstUISettingsProvider
+ *     theme={{
+ *       themeOverrides: {
+ *         canvas: {
+ *           colors: {
+ *             backgroundLightest: '#fefefe'
+ *           },
+ *           borders: {
+ *             style: 'dashed'
+ *           }
+ *         }
+ *       }
+ *     }}
+ *   >
+ *     <div>Canvas with new 'backgroundLightest'</div>
+ *   </InstUISettingsProvider>
+ *
+ *   <InstUISettingsProvider theme={instructure} dir="rtl">
+ *     <div>Instructure themed part with RTL text</div>
+ *   </InstUISettingsProvider>
+ * </InstUISettingsProvider>
+ * ```
+ *
+ * @param {object} children
+ * @param {object} theme - A full theme or an override object
+ * @param {string} dir - The text direction to use in the descendants. If not
+ * given it uses the following in this priority order:
+ *   - The value given in a parent `TextDirectionContext`
+ *   - The `dir` prop of `document.documentElement` or its `direction` CSS prop
+ *   - `ltr`
+ * @module InstUISettingsProvider
+ */
+function InstUISettingsProvider({
+  children,
+  theme = {},
+  dir
+}: React.PropsWithChildren<InstUIProviderProps>) {
+  const finalDir = dir || useContext(TextDirectionContext)
+  return (
+    <ThemeProvider theme={getTheme(theme)}>
+      <TextDirectionContext.Provider value={finalDir}>
+        {children}
+      </TextDirectionContext.Provider>
+    </ThemeProvider>
+  )
+}
+
+InstUISettingsProvider.defaultProps = {
+  theme: {},
+  dir: undefined
+}
```

Comment:
The `dir` property is not being shown in the Properties table, and the text is not looking right either:
<img width="985" alt="Screenshot 2021-11-08 at 10 46 54" src="https://user-images.githubusercontent.com/20640237/140719883-81f807e0-edb8-4b6d-adbf-8c8c3eb59b28.png">

Reply:
I could not fix it after many tries, seems to be a bug in react-docgen and I could not find a way to work around it. I've added the description of the parameters into the main description text.